csi.proto

```diff
@@ -1634,3 +1682,200 @@ message NodeExpandVolumeResponse {
   // The capacity of the volume in bytes. This field is OPTIONAL.
   int64 capacity_bytes = 1;
 }
+message GroupControllerGetCapabilitiesRequest {
+  // Intentionally empty.
+}
+
+message GroupControllerGetCapabilitiesResponse {
+  // All the capabilities that the group controller service supports.
+  // This field is OPTIONAL.
+  repeated GroupControllerServiceCapability capabilities = 1;
+}
+
+// Specifies a capability of the group controller service.
+message GroupControllerServiceCapability {
+  message RPC {
+    enum Type {
+      UNKNOWN = 0;
+
+      // Indicates that the group controller plugin supports
+      // creating and deleting a volume group snapshot.
+      CREATE_DELETE_VOLUME_GROUP_SNAPSHOT = 1
+      [(alpha_enum_value) = true];
+
+      // Indicates that the group controller plugin supports getting
+      // details of a volume group snapshot.
+      GET_VOLUME_GROUP_SNAPSHOT = 2 [(alpha_enum_value) = true];
```

Comment:
I'm not sure there is any benefit in making the `Get` call an optional capability. For previous features, we had to make it optional because the RPC was not part of the original design, but in this case it is, and I can't possibly imagine an implementation that can implement Create/Delete but not Get.

My concern is that by splitting it out and making it optional, we're forcing the sidecar for the feature to not rely on this RPC for correctness, which makes it functionally useless. It would be better to combine this with the Create/Delete capability and tie all 3 RPCs to the same capability. I'm okay with leaving List separate because List RPCs in CSI are genuinely weird and they're not required for any CO that I know of.

Reply:
It's probably reasonable to combine Create/Delete with Get as one capability. Create is idempotent and works as a Get in subsequent calls anyway so I'd think Get can be supported if Create can support idempotency.
If no one else is concerned about consistency with other Get and Create/Delete capabilities, I'll combine them and call it CREATE_DELETE_GET_VOLUME_GROUP_SNAPSHOT.